targettests/Remote-Sim/qvector_init_from_state.cpp

```diff
@@ -77,68 +78,57 @@ int main() {
       counts = cudaq::sample(test_state_param, &state1);
       printCounts(counts);
   }
-
 // CHECK: Passing state created from data as argument (kernel mode)
-// CHECK: 000
-// CHECK: 100
-
 // CHECK: 011
 // CHECK: 111
 
+// CHECK: 000
```

Comment:
Q: Do we need `CHECK-DAG` directive if the output order varies?

Reply:
The output order should not vary, the change in the output is due to the addition of the gate x() in `test_state_param`. Apologies for confusion, I'll change the title of the PR to "improve tests and remove flakes"